launch/launch/actions/timer_action.py

```diff
@@ -147,7 +145,7 @@ def execute(self, context: LaunchContext) -> Optional[List['Action']]:
                     )
                 ),
             ))
-            _event_handler_has_been_installed = True
+            setattr(context, "_TimerAction__event_handler_has_been_installed", True)
```

Comment:
Wouldn't it be better to set this to `False` in the context and then set it to `True` here? Rather than externally adding a private member to the context.

Reply:
Please make this suggested change.

Reply:
I thought about that and did it this way because a TimerAction object is required to know about a LaunchContext in order to function, but a LaunchContext shouldn't have to know about a TimerAction in order to do its job.  Also, by making the name start with __TimerAction_ and using setattr, I prevent it from conflicting with any of the Launch Context's mangled names in the future.

If I add a member to the LaunchContext, what do you recommend I name it?  If I prefix it with `__`  then it may be difficult to access from a TimerAction due to name mangling.  Is single underbar ok?

Reply:
> I thought about that and did it this way because a TimerAction object is required to know about a LaunchContext in order to function, but a LaunchContext shouldn't have to know about a TimerAction in order to do its job.

That's reasonable to me, perhaps we could use something like the context's locals (we could us a global to prevent it from being scoped, see: https://github.com/ros2/launch/blob/34f10c69ea0f666348e11c2e5c7d66e947430730/launch/launch/launch_context.py#L94-L102) to store this information? I suppose that's not much better than just add a member to the context class itself.

After considering your point, I'd be ok with you just leave it as-is.